### ISSUE TITLE
Multi location companies

### ISSUE THREAD

**Issue by vpp** [timeoff-management/application#41]
Consider adding concept of locations into companies.

So it is possible to have different sets of bank holidays within same organisation. E.g. if company has offices in different countries.

Here is a comment of one of our customers:

"... Our company is international, which brings a load of complications concerning different calendars and vacation time rules for each country. To generalize it, being able to define different sets of public holidays and determining which set a person gets is the sort of flexibility that we would need.... "

**Comment by vpp** [comment]
One more person is interesting in this feature.

**Comment by vpp** [comment]
One more person asked for this

**Comment by vpp** [comment]
One more important customer asked for this

> International companies: with offices in different countries with different set of Bank holidays.

**Comment by blueman007** [comment]
Same request here. Maybe it could be solved more easily by attaching the bank holidays to the departments, instead of attaching them to the company ?

**Comment by vpp** [comment]
@blueman007 this is very interesting idea! And the more I think about it the better it sounds.

**Comment by blueman007** [comment]
To add more details about the need ; the initial need mentioned that this issue affects international companies, whereas in fact it's much broader than that as it can affect many national companies too. Simply consider all the countries where the bank holidays are different per region : USA (different per state), Canada (different per province), Switzerland (per Canton), etc. And that's only the countries that I have practical experience with.

So don't think that this issue is only for a few international companies (plus thinking "international" implies bigger businesses with a vacation ERP already in place), but also a lot of smaller, national companies.

Thanks for the consideration.

**Comment by blueman007** [comment]
Hi,
Is is possible to get some insight about de development of this feature ? Because we are holding the deployment of TimeOff in our company until this function is available. 
Thanks.

**Comment by vpp** [comment]
hi @blueman007 I've done some work in tom-41-multilocation to introduce "grouping departments" so account could have "department" that could stand for particular office. But I am not sure about this approach at this moment so put the development on hold. Now I am toying with idea of link multiple "companies" into single account, that way it woud be proper multi office setup with separate settings per office such as timezones and time format, bank holiday sets etc.

To summarise: the development is happening, the issue is I am not sure which approach to bring into master.

**Comment by asidroid** [comment]
Hi, As long as the information is visible for the whole organization and the local companies calendars are visually merged, both ways are solving the issue. However, we could also beta-test your temporary solution and give you some feedback before you decide which is the version you want to push on the master.

**Comment by asidroid** [comment]
Hi! Any news on this feature ? I have tested the multi location branch. Unfortunately I couldn't plug the master branch configured database to this release version. An workaround that I've done on the master was to set each department as a individual company with a different location. 
This works fine but it offers no inter-department visibility.

**Comment by vitas** [comment]
We need that! very useful!
